build-tools/packages/build-tools/src/test/biomeConfig.test.ts

```diff
@@ -0,0 +1,315 @@
+					actual.linter!.include!.length === 1,
+					`expected 1 elements in the array, got ${actual.linter!.include!.length}`,
+				);
+			});
+
+			it("linter.ignore has correct value", async () => {
+				const testFile = path.resolve(testDataPath, "biome/pkg-a/biome.jsonc");
+				const actual = await loadBiomeConfig(testFile);
+
+				assert(actual.linter!.ignore!.includes("ignore-linter/**"));
+				assert(
+					actual.linter!.ignore!.length === 1,
+					`expected 1 elements in the array, got ${actual.linter!.ignore!.length}`,
+				);
+			});
+		});
+	});
+
+	describe("getSettingValuesFromBiomeConfig", async () => {
+		describe("extends from a single config", async () => {
+			// These variables need to be initialized once for all the tests in this describe block. Defining them outside
+			// of the before block causes the tests to be skipped.
+			let testConfig: BiomeConfigResolved;
+			before(async () => {
+				const testFile = path.resolve(testDataPath, "biome/pkg-a/biome.jsonc");
+				testConfig = await loadBiomeConfig(testFile);
+			});
+
+			it("formatter ignore settings are merged with root", async () => {
+				const ignores = await getSettingValuesFromBiomeConfig(
+					testConfig,
+					"formatter",
+					"ignore",
+				);
+				assert(ignores.has("pkg-a-ignore/**"));
+				assert(ignores.has("ignore-formatter/**"));
+				assert(ignores.size === 2, `expected 2 items in the set, got ${ignores.size}`);
+			});
+
+			it("linter ignore settings are merged with root", async () => {
+				const ignores = await getSettingValuesFromBiomeConfig(testConfig, "linter", "ignore");
+				assert(ignores.has("pkg-a-ignore/**"));
+				assert(ignores.has("ignore-linter/**"));
+				assert(ignores.size === 2);
+			});
+
+			it("formatter include settings are merged with root", async () => {
+				const includes = await getSettingValuesFromBiomeConfig(
+					testConfig,
+					"formatter",
+					"include",
+				);
+				assert(includes.has("pkg-a-include/**"));
+				assert(includes.has("include-formatter/**"));
+				assert(includes.size === 2);
+			});
+
+			it("linter include settings are merged with root", async () => {
+				const includes = await getSettingValuesFromBiomeConfig(
+					testConfig,
+					"linter",
+					"include",
+				);
+				assert(includes.has("pkg-a-include/**"));
+				assert(includes.has("include-linter/**"));
+				assert(includes.size === 2);
+			});
+		});
+	});
+
+	describe("getBiomeFormattedFilesFromDirectory", async () => {
+		describe("extends from a single config", async () => {
+			// These variables need to be initialized once for all the tests in this describe block. Defining them outside
+			// of the before block causes the tests to be skipped.
+			let testPath: string;
+			let gitRepo: GitRepo;
+			before(async () => {
+				const repoRoot = await getResolvedFluidRoot(true);
+				gitRepo = new GitRepo(repoRoot);
+				testPath = path.resolve(testDataPath, "biome/pkg-a/");
+			});
+
+			it("returns correct file set", async () => {
```

Comment:
An additional case, that can be added later, is for test to create a local file that is .gitignored. Then that should not appear in the file list.

Reply:
> My guess is that testPath can be a simple const. 

You're right.

> But I am puzzled by the async for the describe blocks.

I think I originally tried declaring stuff in the describe block and made everything async, then when I realized that was causing tests to get missed and switched to `before`, I neglected to remove the async I added previously. Cleaned up.

Reply:
Looks like reply is for other comment -> unresolving.